### PR TITLE
(VELOCITY-1073) Use `app_id` rather than `package`

### PR DESCRIPTION
**Pull request by sschneid**
Especially with marathon, the package name does not always match the
service name; using the `app_id` and failing back to the default.

This is an update to original PR #78.

### CODE REVIEW

**Comment by sschneid** [comment]
@mesosphere-ci retest this please

**Comment by colin-msphere** [comment]
Code block is used twice, can this get rolled into a function?

File: tests/acceptance/test_dcos_package.py [hunk 1]
```diff
@@ -16,7 +16,7 @@ def test_uninstall_package_and_wait():
 def test_install_package_with_json_options():
     install_package_and_wait('chronos', None, 'big-chronos', None, {"chronos": {"cpus": 2}})
     assert get_service_task('marathon', 'big-chronos')['resources']['cpus'] == 2
```
Comment (kensipe) [inline tests/acceptance/test_dcos_package.py]: @colin-msphere the get_service_task is a strange function :)   'marathon' is correct here.

the 1st param is the service, the 2nd is the task-name.   Marathon is what launched it.

File: shakedown/dcos/package.py [hunk 2]
```diff
@@ -100,7 +106,7 @@ def install_package(
         future = now + timeout_sec
 
         while now < future:
-            if shakedown.get_service(package_name):
+            if shakedown.get_service(app_id):
```
Comment (kensipe) [inline shakedown/dcos/package.py]: I don't know why we use app_id at all.   From the perspective of the end user they have zero control over app_id.   package_name 'marathon' and service.name 'marathon-user' as the default is what they have control over.

If we launch multiple marathons, they will need 1) different app_id and 2) different service_names.   however by convention in dcos they are the same.   can we call this service name?

Comment (kensipe) [inline shakedown/dcos/package.py]: I don't know why we use app_id at all.   From the perspective of the end user they have zero control over app_id.   package_name 'marathon' and service.name 'marathon-user' as the default is what they have control over.

If we launch multiple marathons, they will need 1) different app_id and 2) different service_names.   however by convention in dcos they are the same.   can we call this service name?

File: shakedown/dcos/package.py [hunk 3]
```diff
@@ -74,6 +74,12 @@ def install_package(
 
     cosmos = _get_cosmos()
     pkg = cosmos.get_package_version(package_name, package_version)
+    if app_id is None:
+        properties = pkg.config_json().get('properties')
+        if 'service' in properties:
+            app_id = properties['service']['properties']['name']['default']
```
Comment (kensipe) [inline shakedown/dcos/package.py]: there is a danger here... this is not defined as a standard universe package.     This works for marathon.

Comment (kensipe) [inline shakedown/dcos/package.py]: marathon:   https://github.com/mesosphere/universe/blob/version-3.x/repo/packages/M/marathon/4/marathon.json.mustache#L3

marathon-lb: https://github.com/mesosphere/universe/blob/version-3.x/repo/packages/M/marathon-lb/14/marathon.json.mustache#L2

these are 2 different ways of setting the "app_id".   They are NOT controlled by the user.

however:
This is the service name: https://github.com/mesosphere/universe/blob/version-3.x/repo/packages/M/marathon/4/marathon.json.mustache#L42

by convention the ['labels']['DCOS_SERVICE_NAME'] is the same as app_id

This is pulling things out of the config.json which we have no convention for.

**Comment by sschneid** [comment]
Okay -- I'm going to re-work this to use `service_name` rather than `app_id` since what I'm hearing is that `app_id` is only used internally.  :)

**Pull request closed without merging**